collect_app/src/test/java/org/odk/collect/android/widgets/SelectOneWidgetTest.java

```diff
@@ -123,6 +123,22 @@ public void whenAChoiceIsBlank_itIsShownAsBlank() {
         assertThat(innerText(widget.getChoicesList().getChildAt(0)), equalTo(""));
     }
 
+    @Test
+    public void whenAChoiceValueIsNull_itIsShownAsBlank() {
+        SelectChoice selectChoice = new SelectChoice(); // The two arg constructor protects against null values
+        selectChoice.setTextID("1");
+
+        formEntryPrompt = new MockFormEntryPromptBuilder()
+                .withSelectChoices(asList(
+                        selectChoice
+                ))
+                .build();
+
+        SelectOneWidget widget = getWidget();
+        populateRecyclerView(widget);
+        assertThat(innerText(widget.getChoicesList().getChildAt(0)), equalTo(""));
```

Comment:
This test would crash without the code change so that's good. But I still don't think it's asserting something that makes sense.